### PR TITLE
Keyvault unit tests for non standard key sizes. 

### PR DESCRIPTION
**Pull request by gearama**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [ ] Comments in source
- [x] No typos
- [ ] Update changelog
- [x] Not work-in-progress
- [ ] External references or docs updated
- [x] Self review of PR done
- [ ] Any breaking changes?

Test names appear like so 
image

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp:131]: move this to cpp file, otherwise, this instantiation will happen every time this header is included. Looks like it is currently included in 7 cpp files, so we are producing 7 instances of each parametrized test.

File: sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp:132]: nit. maybe use something like `client`

File: sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp:127]: what about adding a little more description here like `"KeySize-"+std::to_string(info.param)` or something which helps reading the console out better.

File: sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp]: done

File: sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp]: done

File: sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp]: param is key size , the name of the test becomes "testname/13" or testname/2048 so it's test with the associated key size

File: sdk/keyvault/azure-security-keyvault-keys/test/ut/key_cryptographic_client_test_live.cpp
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-keys/test/ut/key_cryptographic_client_test_live.cpp:242]: This is neat, where is `GetParam()` defined? Does gtest auto-generate it based on `::testing::Values(13, 55, 233, 987, 1597, 2048),`?

Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-keys/test/ut/key_cryptographic_client_test_live.cpp:242]: Get Param is part of gtest , it deals with reading the parameter generated out of that list 
https://github.com/google/googletest/blob/master/docs/advanced.md#value-parameterized-tests

Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-keys/test/ut/key_cryptographic_client_test_live.cpp:242]: so the idea is you write one test , figure out want would be the variable part of the test, put the values in the list declared int he macro , and then the test runs for each parameter value/ element in the list, 
if there's need for multiple values for one test ( not the case here, but you can imagine file name and path, or an image width and height, etc.) you can pass in the list a struct, or tuple or some other blob, gtest doen't really care. 
When you do GetParam you get the same tuple or blob and then you unpack it in the test to get whatever data is relevant for the run.

File: sdk/keyvault/azure-security-keyvault-keys/test/ut/key_cryptographic_client_test_live.cpp
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-keys/test/ut/key_cryptographic_client_test_live.cpp:290]: Are all these key sizes valid? Is 2048 the max? What about negatives, 0, or larger than 2048?

Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-keys/test/ut/key_cryptographic_client_test_live.cpp]: in the context of the C++ api please define valid, there are some fibonacci sequence numbers, i also added some negatives, and some higher powers of 2. 
When we call create key with the key length provided here, we end up calling an azure API  with the parameters we want (one of them being the key size). 
We receive a response of 200and a key back. 
All we have to go on here for defining valid is "will i be able to pass the value to Azure". i don't think we should care beyond this point, what azure does with the key, we should not start testing Azure keyvault implementation. Since Azure graciously comes back with a key, then i would say it's "valid"

**Review by ahsonkhan — APPROVED** [review]
This looks good to me, but I will leave it up to you to decide whether you want @vhvb1989's sign off before merging or not :)

**Review by vhvb1989 — APPROVED** [review]

**Pull request merged**
